src/MongoDB.Driver/MongoClientSettings.cs

```diff
@@ -1275,22 +1275,40 @@ public override string ToString()
         // internal methods
         internal ClusterKey ToClusterKey()
         {
+            CryptClientSettings cryptClientSettings = null;
+
+            if (_autoEncryptionOptions != null &&
```

Comment:
it's enough to look only at `_autoEncryptionOptions != null` since this is a flag when we start configuring auto encryption

Reply:
This would be behavioral change.
Now we configure the client if `cryptClientSettings  != null` and beforehand it was configured if `_autoEncryptionOptions.FieldX != null` and not if `_autoEncryptionOptions != null`, so preserving that.
Remember that this is the `CryptClientSettings.IsSet` change you've requested.

Reply:
not really, the reason why this condition actually exists initially, because we didn't have fle options for that level. The spec says that auto encryption should be configured if auto options has been created